lib/std/http/Client.zig

```diff
@@ -0,0 +1,142 @@
+const std = @import("../std.zig");
+const assert = std.debug.assert;
+const http = std.http;
+const net = std.net;
+const Client = @This();
+
+allocator: std.mem.Allocator,
+headers: std.ArrayListUnmanaged(u8) = .{},
+active_requests: usize = 0,
+ca_bundle: std.crypto.Certificate.Bundle = .{},
+
+pub const Request = struct {
+    client: *Client,
+    stream: net.Stream,
+    headers: std.ArrayListUnmanaged(u8) = .{},
+    tls_client: std.crypto.tls.Client,
+    protocol: Protocol,
+
+    pub const Protocol = enum { http, https };
```

Comment:
question

Since it's public, can it be extended/replaced?

Reply:
Once the std is supposed to reach some kind of stability, this could explicitly be made a non-exhaustive enum so that we can still add protocols in the future.